File: src/main/java/org/tensorflow/keras/mixin/LayerFunction.java
package org.tensorflow.keras.mixin;

import org.tensorflow.Operand;
import org.tensorflow.op.Ops;
import org.tensorflow.types.family.TType;

@FunctionalInterface
public interface LayerFunction<T extends TType> {
  @SuppressWarnings("unchecked")
  Operand<T> apply(Ops tf, Operand<T>... inputs);
}

File: src/main/scala/org/tensorflow/keras/mixin/LayerFunction.scala
package org.tensorflow.keras.mixin

import org.tensorflow.Operand
import org.tensorflow.op.Ops
import org.tensorflow.types.family.TType

@FunctionalInterface trait LayerFunction[T <: TType] {
  @SuppressWarnings(Array("unchecked")) def apply(tf: Ops, inputs: Operand[T]*): Operand[T]
}
